OpenSim/Common/TimeSeriesTable.h

```diff
@@ -183,11 +222,88 @@ class TimeSeriesTable_ : public DataTable_<double, ETY> {
         *this = std::move(*table);
     }
 
+    /** Get row whose time column is nearest/closest to the given value. 
+
+    \param time Value to search for. 
+    \param restrictToTimeRange When true -- Exception is thrown if the given 
+                               value is out-of-range of the time column. 
+                               When false -- If the given value is less than or 
+                               equal to the first value in the time column, the
+                               row returned is the first row. If the given value
+                               is greater than or equal to the last value in the
+                               time column, the row returned is the last row. 
+                               This operation only returns existing rows and 
+                               does not perform any interpolation. Defaults to
+                               'true'.
+
+    \throws TimeOutOfRange If the given value is out-of-range of time column.
+    \throws EmptyTable If the table is empty.                                 */
+    RowVectorView
+    getRowNear(const double& time,
```

Comment:
In #774, we use `findNearestBefore()`, `findNearestAfter()`, and `findAt()`. The last of these will only return a state if the given time is within a _very_ small tolerance of one that exists in the trajectory